pkg/extensions/tbtc/monitoring_lock.go

```diff
@@ -0,0 +1,41 @@
+package tbtc
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type monitoringLockKey string
+
+func newMonitoringLockKey(
+	depositAddress string,
+	monitoringName string,
+) monitoringLockKey {
+	return monitoringLockKey(fmt.Sprintf(
+		"%v-%v",
+		depositAddress,
+		strings.ReplaceAll(monitoringName, " ", ""),
+	))
+}
+
+type monitoringLock struct {
+	locks sync.Map
```

Comment:
Wouldn't it be enough to have `monitoringLocks sync.Map` in `type tbtc struct` and have `tryLock` and `releaseLock` defined on this struct? We'd be able to have exactly the same tests as in `monitoring_lock_test.go`, just one level of indirection less.